base/path.jl

```diff
@@ -246,8 +246,70 @@ function splitpath(p::String)
     return out
 end
 
+joinpath(path::AbstractString) = path
+
+if Sys.iswindows()
+
+function joinpath(path::AbstractString, paths::AbstractString...)
+    result_drive, result_path = splitdrive(path)
+
+    local p_drive, p_path
+    for p in paths
+        p_drive, p_path = splitdrive(p)
+
+        if startswith(p_path, ('\\', '/'))
+            # second path is absolute
+            if !isempty(p_drive) || !isempty(result_drive)
+                result_drive = p_drive
+            end
+            result_path = p_path
+            continue
+        elseif !isempty(p_drive) && p_drive != result_drive
+            if lowercase(p_drive) != lowercase(result_drive)
+                # different drives, ignore the first path entirely
+                result_drive = p_drive
+                result_path = p_path
+                continue
+            end
+            # same drive in different case
+            result_drive = p_drive
```

Comment:
Does it make sense to normalize the drive name capitalization?

Reply:
Are UNC drives also case insensitive?

Reply:
> Does it make sense to normalize the drive name capitalization?

Don't think so. We should just preserve the input capitalization. 

> Are UNC drives also case insensitive?

In this case it doesn't matter since if the user specifies a UNC path it is absolute.